projects/wave-core-new/src/lib/datatable/table/full-display/full-display.component.ts

```diff
@@ -0,0 +1,63 @@
+import {Component, OnInit, ChangeDetectionStrategy, Inject} from '@angular/core';
+import {MAT_DIALOG_DATA} from '@angular/material/dialog';
+import {Clipboard} from '@angular/cdk/clipboard';
+import {Geometry, MultiPolygon} from 'ol/geom';
+
+/**
+ * Opened as modal dialog to display a full set of coordinates and allow copying to clipboard
+ */
+@Component({
+    selector: 'wave-full-display',
+    templateUrl: './full-display.component.html',
+    styleUrls: ['./full-display.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class FullDisplayComponent implements OnInit {
+    xCoords: string[] = [];
+    yCoords: string[] = [];
+
+    constructor(
+        @Inject(MAT_DIALOG_DATA) public data: {xStrings: string[]; yStrings: string[]; geometry: Geometry},
+        private clipboard: Clipboard,
+    ) {}
+
+    ngOnInit(): void {
+        this.xCoords = this.data.xStrings;
+        this.yCoords = this.data.yStrings;
+    }
+
+    copyToClipboard(): void {
+        const type = this.data.geometry.getType();
+        const prettyCoords = JSON.stringify((this.data.geometry as MultiPolygon).getCoordinates(), null, 2);
+        const result = `{ "type": "${type}",\n  "coordinates":\n    ${prettyCoords}\n}`;
```

Comment:
I would prefer using openlayers methods for that. This should be more robust:

```ts
import OlFormatGeoJson from 'ol/format/GeoJSON';

[…]

const geometryJson = JSON.stringify(new OlFormatGeoJson().writeGeometryObject(this.data.geometry), null, 2);
console.log(geometryJson);
```

Reply:
It's using the openlayers method now c5fa8fea182296357d055b3a755854320a526ab7